### PR TITLE
Fixed incorrect function signatures in VectorList and added some tests

### PR DESCRIPTION
**Pull request by thomasfanell**
* Adapted ```VectorList map()``` for rock 1.0.11
* Added test for ```VectorList sort()```
* Style fix in ```VectorList map()```

### CODE REVIEW

File: test/collections/VectorListTest.ooc [hunk 1]
```diff
@@ -225,7 +225,13 @@ VectorListTest: class extends Fixture {
 			list free()
 		})
 		this add("VectorList sort", func {
-			//TODO Current way of sorting not supported by Rock
+			list := VectorList<Int> new()
+			list add(8); list add(16); list add(32)
+			sortedList := list copy()
+			sortedList sort(|v1, v2| v1 < v2)
```
Comment (davidhesselbom) [inline test/collections/VectorListTest.ooc]: Can you try to create a closure that does this instead? I'm pretty sure `| ... |` will cause a memory leak, in which case this isn't how we need to be doing it where such things matter (i.e. outside of tests).

**Comment by marcusnaslund** [comment]
So sort will work now, but not fold?

File: test/collections/VectorListTest.ooc (hunk 1 above)
Comment (thomasfanell) [inline test/collections/VectorListTest.ooc]:
```ooc
sortedList sort(func(left, right: Int) -> Bool { left < right })
```
fails with
```ooc
call sortedList sort((left: Int, right: Int) -> Bool) waiting on VectorList sort(greaterThan: Func (T, T) -> Bool) to resolve.
```

Comment (marcusnaslund) [inline test/collections/VectorListTest.ooc]: What about
```ooc
sortFunc: Func (left, right: Int) -> Bool { left < right }
sortedList sort(sortFunc)
```
or even
```
```ooc
sortFunc: Func (left, right: Int) -> Bool { left < right ? true : false }
sortedList sort(sortFunc)
```

File: source/collections/VectorList.ooc [hunk 2]
```diff
@@ -123,11 +123,10 @@ VectorList: class <T> {
...
-	map: func <T, S> (function: Func(T) -> S) -> This<S> {
+	map: func <S> (function: Func(T) -> S) -> This<S> {
 		result := This<S> new(this count)
-		for (i in 0 .. this count) {
+		for (i in 0 .. this count)
 			result add(function(this[i]))
-		}
 		result
 	}
 	fold: func <T, S> (function: Func(T, S) -> S, initial: S) -> S {
```
Comment (marcusnaslund) [inline source/collections/VectorList.ooc]: `func <S>` here as well then?

File: test/collections/VectorListTest.ooc (hunk 1 above)
Comment (davidhesselbom) [inline test/collections/VectorListTest.ooc]: Well then, I guess https://github.com/fasterthanlime/rock/issues/923 is still relevant.

Comment (thomasfanell) [inline test/collections/VectorListTest.ooc]: @marcusnaslund same

File: source/collections/VectorList.ooc (hunk 2 above)
Comment (thomasfanell) [inline source/collections/VectorList.ooc]: It's still the same problem: ```error Not enough info to resolve return type S of function call```. Which could be solved by qualifying ```S: Class```, but that is not acceptable in my opinion.

Comment (thomasfanell) [inline source/collections/VectorList.ooc]: Anyway, the rock-devs are checking out some issues as we speak. Let's wait a day or two and see what they come up with.

Comment (thomasfanell) [inline source/collections/VectorList.ooc]: So it appears that we for now have to live with the fact that we must qualify ```S``` under certain circumstances, such as when ```S``` is used in an argument in a ```Func```, as is the case with ```VectorList fold()```.

Comment (marcusnaslund) [inline source/collections/VectorList.ooc]: If that *works*, it's better than before, which didn't generally work at all, even if you specified `S`. Is that the case now?

File: test/collections/VectorListTest.ooc (hunk 1 above)
Comment (marcusnaslund) [inline test/collections/VectorListTest.ooc]: Still true?

File: test/collections/VectorListTest.ooc [hunk 3]
```diff
@@ -225,10 +225,19 @@ VectorListTest: class extends Fixture {
 			list free()
 		})
 		this add("VectorList sort", func {
-			//TODO Current way of sorting not supported by Rock
+			list := VectorList<Int> new()
+			list add(8); list add(16); list add(32)
```
Comment (marcusnaslund) [inline test/collections/VectorListTest.ooc]: This isn't consistent with the above tests, but please do `list add(8) . add(16) . add(32)` instead.

File: test/collections/VectorListTest.ooc [hunk 4]
```diff
@@ -225,10 +225,19 @@ VectorListTest: class extends Fixture {
...
+			sortedList := list copy()
+			sortedList sort(|v1, v2| v1 < v2)
+			count := list count
+			for (value in sortedList)
+				expect(value, is equal to(list[--count]))
 		})
 		this add("VectorList fold", func {
-			//TODO Current way of folding not supported by Rock
+			list := VectorList<Int> new()
+			list add(1); list add(2); list add(3)
```
Comment (marcusnaslund) [inline test/collections/VectorListTest.ooc]: `list add(1) . add(2) . add(3)`

**Comment by davidhesselbom** [comment]
So... all good to go?

**Comment by thomasfanell** [comment]
Yup :+1:

**Comment by marcusnaslund** [comment]
I was expecting something to fail without this, but that's not the case.

**Comment by thomasfanell** [comment]
That was the case before we changed ```func <S>```, since rock segfaulted when we attempted to redefine ```T```. I included the segfault fix in the latest rock version, so the title of this PR is now obsolete.

**Comment by marcusnaslund** [comment]
Validated @davidhesselbom

**Comment by marcusnaslund** [comment]
I'm happy with this, it passes validation, and there are no side-effects. @davidhesselbom

**Pull request merged**